crates/adapters/src/transport/datagen.rs

```diff
@@ -924,7 +1024,15 @@ impl RecordGenerator {
                 (DatagenStrategy::Zipf, Some(values)) => {
                     let zipf = Zipf::new(values.len() as u64, settings.e as f64).unwrap();
                     let idx = rng.sample(zipf) as usize - 1;
-                    *obj = values[idx].clone();
+                    let new_value = values[idx].clone();
+                    if !matches!(new_value, Value::String(_)) {
```

Comment:
this error check is the same everywhere, in another language you could write it only once

Reply:
fixed